javaagent-tooling/javaagent-tooling-java9/src/main/java/io/opentelemetry/javaagent/tooling/ModuleOpener.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import static java.util.logging.Level.FINE;
+import static java.util.logging.Level.WARNING;
+
+import java.lang.instrument.Instrumentation;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+public class ModuleOpener {
+
+  private static final Logger logger = Logger.getLogger(ModuleOpener.class.getName());
+
+  private ModuleOpener() {}
+
+  /**
+   * Opens JPMS module to a class loader unnamed module
+   *
+   * @param classFromTargetModule class from target module
+   * @param openTo class loader to open module for
+   * @param packagesToOpen packages to open
+   */
+  public static void open(
+      Instrumentation instrumentation,
+      Class<?> classFromTargetModule,
+      ClassLoader openTo,
+      Collection<String> packagesToOpen) {
+
+    Module targetModule = classFromTargetModule.getModule();
+    Module openToModule = openTo.getUnnamedModule();
+    Set<Module> openToModuleSet = Collections.singleton(openToModule);
+    Map<String, Set<Module>> missingOpens = new HashMap<>();
+    for (String packageName : packagesToOpen) {
+      if (!targetModule.isOpen(packageName, openToModule)) {
+        missingOpens.put(packageName, openToModuleSet);
+        logger.log(
+            FINE,
+            () ->
+                String.format(
+                    "Exposing package '%s' in module '%s' to module '%s'",
+                    packageName, targetModule, openToModule));
+      }
+    }
+    if (missingOpens.isEmpty()) {
+      return;
+    }
+
+    if (!instrumentation.isModifiableModule(targetModule)) {
+      logger.log(WARNING, "Module '{}' can't be modified", targetModule);
```

Comment:
IINM JUL doesn't accept `{}` as template and requires explicit indices `{0}`, right?